sbin/send_notifications.py

```diff
@@ -0,0 +1,201 @@
+            if len(notification_list) > 0:
+                course_db.execute(
+                    f"""INSERT INTO notifications
+                    (component, metadata, content, created_at, from_user_id,
+                    to_user_id)
+                    VALUES {", ".join(notification_list)};"""
+                )
+
+            # Send out emails using both course and master database
+            email_list = []
+            email_subject = (f"[Submitty {course}] Grade Released: "
+                             f"{gradeable['title']}")
+
+            email_body = (f"An Instructor has released scores in:\n{course}"
+                          f"\n\nScores have been released for "
+                          f"{gradeable['title']}.\n\nAuthor: System\n"
+                          f"Click here for more info: {gradeable_url}\n\n"
+                          "--\n"
+                          "NOTE: This is an automated email notification, "
+                          "which is unable to receive replies.\n"
+                          "Please refer to the course syllabus for contact "
+                          "information for your teaching staff.")
+
+            email_recipients = course_db.execute(
+                """
+                SELECT users.user_id , users.user_email
+                FROM users
+                JOIN notification_settings
+                ON notification_settings.user_id = users.user_id
+                WHERE notification_settings.all_released_grades_email = true;
+            """
+            )
+
+            for recipient in email_recipients:
+                user_id, user_email = recipient[0], recipient[1]
+                email_list.append(
+                    f"('{email_subject}', '{email_body}', '{timestamp}', "
+                    f"'{user_id}', '{user_email}', '{term}', '{course}')"
+                )
+
+            if len(email_list) > 0:
+                master_db.execute(
+                    f"""INSERT INTO emails
+                    (subject, body, created, user_id, email_address, term,
+                    course)
+                    VALUES {", ".join(email_list)};"""
+                )
+
+            # Add successfully notified gradeables to update state
+            notified_gradeables.append(f"'{gradeable['id']}'")
+
+        # Update all successfully sent notifications for current course
+        if len(notified_gradeables) > 0:
+            course_db.execute(
+                f"""UPDATE gradeable SET g_notification_state = true
+                WHERE g_id in ({", ".join(notified_gradeables)})"""
+            )
+            total_notified_gradeables += 1
+
+        # Close the course database connection
+        course_db.close()
+    return total_notified_gradeables
+
+
+def main():
+    """Driver method to release pending notifications for gradeables"""
+    try:
+        notified = notify_pending_gradeables()
+        m = (f"[{datetime.datetime.now()}] Successfully released {notified} "
```

Comment:
It might be useful in the logs to include the total number of notifications/emails that were sent due to each run.